pkg/helm/updates.go

```diff
@@ -171,10 +179,58 @@ func PullChartVersion(helmApp *apptypes.HelmApp, licenseID string, version strin
 	}
 
 	imageName := fmt.Sprintf("%s:%s", helmApp.ChartPath, version)
-	data, err := chartGetter.Get(imageName)
+	data, err = chartGetter.Get(imageName)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get chart %q", imageName)
 	}
 
+	data, _ = saveUpdateChartInCache(helmApp, version, data)
+
 	return data, nil
 }
+
+var (
+	updateCacheDir = ""
+)
+
+func getUpdateChartFromCache(helmApp *apptypes.HelmApp, version string) (*bytes.Buffer, error) {
+	fileName := getUpdateChacheFileName(helmApp, version)
+	b, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, errors.Wrap(err, "failed to read file")
+	}
+
+	return bytes.NewBuffer(b), nil
+}
+
+func saveUpdateChartInCache(helmApp *apptypes.HelmApp, version string, data *bytes.Buffer) (*bytes.Buffer, error) {
+	if updateCacheDir == "" {
+		dirName, err := ioutil.TempDir("", "chart-updates-")
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create temp dir")
+		}
+		updateCacheDir = dirName
```

Comment:
so `updateCacheDir` will be lost if kots gets rebooted..then kots will create a new cache dir causing a memory leak..can we add a cleanup defer statement for the cache dir when kots is rebooted?

Reply:
it's in the ephemeral storage.  it will be deleted kotsadm is restarted